Windows: Handle missing _MM_SESSION_SPACE

dgmcdona opened the pull request:
refs #1385

As of Windows 11 24H2, the `_MM_SESSION_SPACE` type no longer appears in the kernel PDB. Instead, the `_EPROCESS.Session` member refers to a new, opaque type, `_PSP_SESSION_SPACE`. However, experimentation has shown that this new structure is functionally identical to the old structure - the `ProcessList` and `SessionId` members still appear to be at their old offsets. In order to account for this when analyzing these newer Windows versions, this catches the `SymbolError` and instantiates an `unsigned long` at the offset (8) where the `SessionId` member would normally be defined within an `_MM_SESSION_SPACE` structure.

This fixes plugins that rely on this member, including `windows.pslist` and `windows.verinfo`

@ikelos I considered adding a new symbol table with this `_PSP_SESSION_SPACE` structure, but this was the quick and easy solution - let me know if you'd prefer an approach like that instead.

dgmcdona committed:
Windows: Handle missing _MM_SESSION_SPACE

As of Windows 11 24H2, the `_MM_SESSION_SPACE` type no longer appears in
the kernel PDB. Instead, the `_EPROCESS.Session` member refers to a new
type, `_PSP_SESSION_SPACE`, which does not have a type definition.
However, experimentation has shown that this new structure is
functionally identical to the old structure - the `ProcessList`
and `SessionId` members still appear to be at their old offsets.
In order to account for this when analyzing these newer Windows
versions, this catches the `SymbolError` and instantiates an `unsigned
long` at the offset (8) where the `SessionId` member would normally be
defined within an `_MM_SESSION_SPACE` structure.

ikelos commented on volatility3/framework/plugins/windows/modules.py:
```diff
@@ -165,13 +165,33 @@ def get_session_layers(
+                    # stores its session ID at offset 8 as an unsigned long, we
+                    # create an unsigned long at that offset and use that
+                    # instead.
+                    session_id = int(
```
This shouldn't need coercing to an int, is there a reason for doing this?  Since the base types should be based off of normal ints, comparison and equality (for example, for set membership) *should* all work normally (but please check, if it doesn't it might be better to fix that for the whole base type rather than doing casts).

dgmcdona replied:
Fixed, added a type hint to the method signature as well.

ikelos reviewed (commented):
Looks reasonable, just want to make sure we're casting to `int` for a reason, otherwise it'll set a bad precedent (and perpetuate people's distrust of the volatility object system).

dgmcdona committed:
Windows: Typing - Remove type casts, add signature

Removes the needless `int` casts, and adds the return type to the
`get_session_id` method signature.
